tool-openssl/tool_util.cc

```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "internal.h"
+#include <string>
+#if !defined(OPENSSL_WINDOWS)
+  #include <sys/stat.h>
+#endif
+
+static bool isCharUpperCaseEqual(char a, char b) {
+  return ::toupper(a) == ::toupper(b);
+}
+
+bool isStringUpperCaseEqual(const std::string &a, const std::string &b) {
+  return a.size() == b.size() &&
+         std::equal(a.begin(), a.end(), b.begin(), isCharUpperCaseEqual);
+}
+
+void SetUmaskForPrivateKey() {
```

Comment:
i confirmed that umask always succeeds on linux, so void return type here is fine

https://man7.org/linux/man-pages/man2/umask.2.html